### PR TITLE
core(full-page-screenshot): handle ShadowRoots

### PR DESCRIPTION
**Pull request by brendankenny**
if you look at [existing deployments](https://lighthouse-hkrvqczsz.vercel.app/english/), you'll notice the `shadow_v2` screenshot bounding boxes are a little weird. Trying to `yarn update:sample-artifacts FullPageScreenshot`, however, will give a gatherer error (`element.getBoundingClientRect is not a function`).

Turns out we were trying to `ShadowRoot.getBoundingRect`, which isn't a thing. FullPageScreenshot just needs to use the `host`, like `getNodeDetails` does when taking the original rect.

We didn't notice because #11536 didn't update the artifact, and the screenshot renderer falls back the audits' bounding rects if there isn't one bundled with the screenshot:

https://github.com/GoogleChrome/lighthouse/blob/966a206cb3c4fd2317788c956b6007064e7c33e5/lighthouse-core/report/html/renderer/details-renderer.js#L526-L527

and since dbw_tester gets so weird with an extended height while taking the screenshot, the the original bounding rects on the full screenshot were pretty little weird :)

### CODE REVIEW

File: lighthouse-cli/test/smokehouse/test-definitions/dobetterweb/dbw-expectations.js [hunk 1]
```diff
@@ -429,6 +429,28 @@ const expectations = [
...
+            nodes: {
+              'page-0-IMG': {
+                top: 639,
+                bottom: 654,
+                left: 8,
+                right: 128,
+                width: 120,
+                height: 15,
+              },
+              // And then many more nodes.
```
Comment (brendankenny) [inline lighthouse-cli/test/smokehouse/test-definitions/dobetterweb/dbw-expectations.js:450]: open to expectation suggestions :)

File: lighthouse-core/lib/page-functions.js [hunk 2]
```diff
@@ -467,26 +467,27 @@ function getNodeDetailsImpl(element) {
...
     lhId = [
       window.__lighthouseExecutionContextId !== undefined ?
         window.__lighthouseExecutionContextId :
         'page',
       window.__lighthouseNodesDontTouchOrAllVarianceGoesAway.size,
-      element.tagName,
+      htmlElement.tagName,
     ].join('-');
-    window.__lighthouseNodesDontTouchOrAllVarianceGoesAway.set(element, lhId);
+    window.__lighthouseNodesDontTouchOrAllVarianceGoesAway.set(htmlElement, lhId);
```
Comment (brendankenny) [inline lighthouse-core/lib/page-functions.js:488]: the other option is to do another ShadowRoot check here:

https://github.com/GoogleChrome/lighthouse/blob/966a206cb3c4fd2317788c956b6007064e7c33e5/lighthouse-core/gather/gatherers/full-page-screenshot.js#L88

but this seems in keeping with the intent of `getNodeDetails` (see the `getBoundingClientRect(htmlElement)`, below)

**Comment by connorjclark** [comment]
LGTM

this didn't change in this PR but I noticed this:

image

weird... that is highlighting an image, not a `<p>`.

**Review by connorjclark — APPROVED** [review]

**Review by paulirish — APPROVED** [review]

**Comment by brendankenny** [comment]
> this didn't change in this PR but I noticed this:
> 
> image
> 
> weird... that is highlighting an image, not a `<p>`.

yeah, this is actually really weird. That text ("This domain is for use in illustrative examples in documents...") is from example.com 🙃 

Is the saved trace for sample_v2 mixed up somehow?

**Comment by brendankenny** [comment]
or wait, only the element reference comes from the trace. `nodeDetails` for `TraceElements` is where it originated:

https://github.com/GoogleChrome/lighthouse/blob/966a206cb3c4fd2317788c956b6007064e7c33e5/lighthouse-core/test/results/artifacts/artifacts.json#L3053-L3070

I don't see anywhere in `dbw_tester` is could be including that snippet...

**Comment by connorjclark** [comment]
I believe it was manually added https://github.com/GoogleChrome/lighthouse/pull/10517/files#diff-fccf071866992ae51684ed3c5edc57deda156a9ef8bbf41caeeeb5ba8931d205R2086

my b. I didnt know about `yarn update:sample-artifacts X` so I must have advised to edit that file manually.

wanna rebaseline that artifact?

**Comment by brendankenny** [comment]
> wanna rebaseline that artifact?

ok, I sort of rebaselined it. I updated the LCP trace element (because I think a lot of people are looking to that for checking that the fp screenshot is working), but fully rebaselining removes the cls and animation elements, because there aren't any on the page (apparently). So I kept those manually added ones so `sample_v2` still makes an interesting deployment page.

We can figure out some sort of policy on that some other day :)

**Pull request merged**